src/renderer/drawers/Provider.js

```diff
@@ -0,0 +1,51 @@
+/* @flow */
+import React, { useReducer } from "react";
+import { SideDrawer } from "~/renderer/components/SideDrawer";
+
+type State = {
+  Component: *,
+  props: *,
+  open: boolean,
+};
+
+// actions
+// it makes them available and current from connector events handlers
+export let setDrawer: (Component?: *, props?: *) => void = () => {};
```

Comment:
i dont get the use of this here :thinking: you want to be able to control the content outside of the Provider context?
i mean why use a context if you're not using it :sweat_smile:

Reply:
i am using it. at the root of the app. instead of having a map key => component like we do for modal, i just passed directly the component

Reply:
i've updated, extracted the drawer component, its now using the context

Reply:
more generally, i've discovered that recently, i prefer to import the action than passing it in the context